xonsh/aliases.py

```diff
@@ -254,7 +254,8 @@ def source_cmd(args, stdin=None):
     args[0] = fpath if fpath else args[0]
     if not os.path.isfile(args[0]):
         raise FileNotFoundError(args[0])
-    prevcmd = 'call "{}"\necho off'.format(args[0])
+    prevcmd = 'call "{}"'.format('" "'.join(args))
```

Comment:
This is probably a very bad way of escaping. I've had to deal with CMD escaping and it encourages people to use `^` for all meta characters instead of quoting them.